src/Contrib/Jaeger/HttpCollectorExporter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use InvalidArgumentException;
+use OpenTelemetry\SDK\Trace\Behavior\SpanExporterTrait;
+use OpenTelemetry\SDK\Trace\Behavior\UsesSpanConverterTrait;
+use OpenTelemetry\SDK\Trace\SpanExporterInterface;
+
+class HttpCollectorExporter implements SpanExporterInterface
+{
+    use UsesSpanConverterTrait;
+    use SpanExporterTrait;
+
+    private SpanConverter $spanConverter;
+
+    private ThriftHttpSender $sender;
+
+    public function __construct(
+        $name,
+        string $endpointUrl
+    ) {
+        $parsedDsn = parse_url($endpointUrl);
+
+        if (!is_array($parsedDsn)) {
+            throw new InvalidArgumentException('Unable to parse provided DSN');
+        }
+
+        if (!isset($parsedDsn['host']) || !isset($parsedDsn['port'])) {
+            throw new InvalidArgumentException('Endpoint should have host, port');
```

Comment:
Why does the endpoint need to have a port?

Reply:
It looks like THttpClient uses it in the exception messages it's "read" method can throw, and then also in its "flush" method (here's a [link to that file](https://github.com/apache/thrift/blob/master/lib/php/lib/Transport/THttpClient.php#L178))

I've currently got that in-progress work to override the "flush" method and replace it entirely with the PSR-18/-7 implementation, which doesn't need to use it, so it's only the exceptions using it now.

I did notice that [Evernote's SDK's usage](https://github.com/evernote/evernote-sdk-php/blob/master/lib/Evernote/Client.php#L163) of THTTPClient went ahead and just inferred the ports from whether the scheme was http or https (so 80 or 443),.

I went ahead just now and did the same here since that seemed reasonable to me rather than requiring callers to specify it